Automattic/jetpack
# Boost: Track Critical CSS generation Statistics

Fixes #20974

#### Description
Changes in this pull request allows us to track the following:

- The size of Critical CSS blocks which are generated,
- The length of time it takes to generate all the Critical CSS for a site,
- How often each type of error comes up,
- How often the process succeeds.

#### Changes proposed in this Pull Request:
- Enqueue tracks javascript library
- Add callback for tracking events triggered via ajax calls.
- Record the starting time of _Generate Critical CSS_ process
- Send the event to Tracks after completion of each Critical CSS block (for both success and failure).

#### Jetpack product discussion


NA

#### Does this pull request change what data or activity we track or use?


Yes, the following:
- Time taken for _Generate Critical CSS_ process
- Status of _Generate Critical CSS_ process (`success` or `error`)
- Error type and Error message in case of Generate Critical CSS fails.
- Percentage of the process completed in case the _Generate Critical CSS_ fails before completion.

#### Testing instructions:






* Go to a site with Jetpack Boost Plugin activated.
* Navigate to wp-admin/admin.php?page=jetpack-boost.
* Enable `Optimize CSS Loading` module.
* Once the process completes, navigate to the Tracks dashboard and verify the recorded event as explained here: PCYsg-4D1-p2
* Note: Usually events take about 5 minutes to reflect on the dashboard.
<img width="911" alt="tracks" src="https://user-images.githubusercontent.com/8727183/133391841-ca1e3d3c-2c26-499d-a3db-366360cc2c59.png">
